pkg/filesystem/subvolume.go

```diff
@@ -125,12 +125,27 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	subvolumeNames := make(map[string]subVolumeInfo)
 	for _, pv := range pvList.Items {
 		if pv.Spec.CSI != nil {
-			subvolumeNames[pv.Spec.CSI.VolumeAttributes["subvolumeName"]] = subVolumeInfo{}
+			subvolumePath := pv.Spec.CSI.VolumeAttributes["subvolumePath"]
+			name := getSubvolumeNameFromPath(subvolumePath)
+			subvolumeNames[name] = subVolumeInfo{}
 		}
 	}
 	return subvolumeNames
 }
 
+// getSubvolumeNameFromPath get the subvolumename from the path.
+// subvolumepath: /volumes/csi/csi-vol-6a99b552-fdcc-441d-b1e6-a522a85a503d/5f4e4caa-f835-41ba-83c1-5bbd57f6aedf
+// subvolumename: csi-vol-6a99b552-fdcc-441d-b1e6-a522a85a503d
+func getSubvolumeNameFromPath(path string) string {
+	splitSubvol := strings.Split(path, "/")
+	if len(splitSubvol) < 3 {
```

Comment:
If you're accessing `splitSubvol[3]` below, then we need at least 4 entries to avoid an index exception
```suggestion
	if len(splitSubvol) < 4 {
```